content/master/guides/composition-functions.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Composition Function
+state: alpha
+---
+
+Composition Functions allow you to supplement or replace your Compositions with
+advanced logic, using either programming languages such as Go or Python, or
+relevant tools such as Helm, kustomize, or CUE. They compliment contemporary
+"Patch and Transform" (P&T) style Composition - it's possible to use only P&T,
+only Functions, or a mix of both in the same Composition.
+
+```yaml
+apiVersion: apiextensions.crossplane.io/v2alpha1
+kind: Composition
+metadata:
+  name: example
+spec:
+  compositeTypeRef:
+    apiVersion: database.example.org/v1alpha1
+    kind: XPostgreSQLInstance
+  functions:
+  - name: my-cool-function
+    type: Container
+    container:
+      image: xpkg.upbound.io/my-cool-function:0.1.0
+```
+
+A Composition Function is a short-lived OCI container that tells Crossplane how
+to reconcile a Composite Resource (XR). The preceding example shows a minimal
+`Composition` that uses a Composition Function rather than the typical P&T style
+array of `resources`.
+
+## Enabling Composition Functions
+
+To enable support for Composition Functions you must:
+
+ * Enable the alpha feature flag.
+ * Deploy a runner that is responsible for running functions.
+
+```console
+# TODO(negz): Update this to use crossplane-stable, not an RC build, once v1.11 has been released.
+
+kubectl create namespace crossplane-system
+
+helm repo add crossplane-master https://charts.crossplane.io/master/
+helm repo update
+
+# Find the latest RC build of Crossplane - e.g. 1.11.0-rc.0.274.g51a07e50
+helm search repo crossplane-master/crossplane --devel
+
+RC_VERSION="v1.11.0-rc.0.274.g51a07e50"
+helm install crossplane --namespace crossplane-system crossplane-master/crossplane \
+    --devel --version ${RC_VERSION} \
+    --set "args={--debug,--enable-composition-functions}" \
+    --set "xfn.enabled=true" \
+    --set "xfn.args={--debug}"
+```
+
+The preceding Helm command installs Crossplane with the Composition Functions
+feature flag enabled, and with the reference "`xfn`" Composition Function runner
+deployed as a sidecar pod. You can confirm Composition Functions are enabled by
+looking for a log line like:
+
+```json
+{"level":"info","ts":1674535093.36186,"logger":"crossplane","msg":"Alpha feature enabled","flag":"EnableAlphaCompositionFunctions"}
+```
+
+## The xfn Runner
+
+Composition Function runners are designed to be pluggable. Each time Crossplane
+needs to invoke a Composition Function it makes a gRPC call to a configurable
+endpoint. The default, reference Composition Function runner is named `xfn`.
+
+![Crossplane talking to xfn via gRPC](composition-functions-xfn-runner.png)
```

Comment:
Can you use the `{{<img >}}` shortcode to optimize the file and make it responsive? https://docs.crossplane.io/v1.10/contributing/docs/#images